pkg/generator/caches_test.go

```diff
@@ -212,8 +212,8 @@ func TestValidateIngress(t *testing.T) {
 		internalVirtualHosts: []*route.VirtualHost{{Name: "internal_host_for_ingress_2", Domains: []string{"internal_host_for_ingress_1"}}},
 	}
 
-	err := caches.ValidateIngress(&translatedIngress)
-	if err == nil {
+	err := caches.validateIngress(&translatedIngress)
+	if err == ErrDomainConflict {
```

Comment:
This should be `!=` right? We could also use `assert.Error(t, err, ErrDomainConflict)`